scenarios/remote_signals/src/main.rs

```diff
@@ -42,122 +40,110 @@ impl Default for ScenarioValues {
 
 impl UserValuesConstraint for ScenarioValues {}
 
+fn setup(ctx: &mut RunnerContext<HolochainRunnerContext>) -> HookResult {
+    configure_app_ws_url(ctx)?;
+    Ok(())
+}
+
 fn agent_setup(
-    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
 ) -> HookResult {
-    connect_trycp_client(ctx)?;
-    reset_trycp_remote(ctx)?;
+    install_app(
+        ctx,
+        scenario_happ_path!("remote_signal"),
+        &"remote_signal".to_string(),
+    )?;
 
     let pending_set = ctx.get().scenario_values.pending_set.clone();
-
-    let client = ctx.get().trycp_client();
-    let agent_name = ctx.agent_name().to_string();
     let reporter = ctx.runner_context().reporter();
+    let agent_index = ctx.agent_index();
+    let client = ctx.get().app_client();
 
     ctx.runner_context()
         .executor()
         .execute_in_place(async move {
-            client
-                .configure_player(agent_name.clone(), conductor_config().to_string(), None)
-                .await?;
-
-            client.startup(agent_name.clone(), None).await?;
-
             let _rcv_task = tokio::task::spawn(async move {
-                loop {
-                    let msg: Signal = ExternIO(
-                        client
-                            .recv_signal()
-                            .await
-                            .expect("signal receiver ended")
-                            .data,
-                    )
-                    .decode()
-                    .unwrap();
-                    let msg = match msg {
-                        Signal::App { signal, .. } => {
-                            let msg: Vec<u8> = signal.into_inner().decode().unwrap();
-                            let msg: TimedMessage = ExternIO(msg).decode().unwrap();
-                            msg
-                        }
-                        _ => continue,
-                    };
-
-                    pending_set.lock().unwrap().remove(&msg.to_request());
-
-                    match msg {
-                        TimedMessage::TimedRequest { .. } => (),
-                        TimedMessage::TimedResponse { requested_at, .. } => {
-                            let dispatch_time_s = requested_at.as_micros() as f64 / 1_000_000.0;
-                            let receive_time_s = Timestamp::now().as_micros() as f64 / 1_000_000.0;
-
-                            reporter.add_custom(
-                                ReportMetric::new("remote_signal_round_trip")
-                                    .with_field("value", receive_time_s - dispatch_time_s),
-                            );
+                // loop {
+                client
+                    .on_signal(move |signal| {
+                        println!("agent {:?} received a signal", agent_index);
```

Comment:
Prefer `tracing` if this is useful, it's easier to filter out information you don't need

Reply:
I knew there was a reason why I had this as draft =D

Reply:
Now it's ready